### PR TITLE
Update README

### LINKED ISSUE #237: Update README
Update the README with the following:
- Summary text to include new components introduced in v2 as well as mentioning that this is v2
- Below summary paragraph add "The v1 version of IAI MovieBot has been presented as a demonstration paper at CIKM'20 [PDF], while the v2 version is to appear at WSDM'24 [PDF]." with links to PDFs on arXiv
- Add "Components" section including the overview figure from paper (keep the color-coding of the new components) and an itemized list of the main components on the figure; for each component we should link the appropriate page within the documentation (issues to be created for those that don't exist) and also list the specific models, where applicable
- Add screenshot from newest version and add links to web demo (separate issues may be created for these)
- Versions section with link to latest v1 release, should anyone need that
- Rename the Publication section to Citation and "For the most recent version of IAI MovieBot, please cite" and WSDM'24 paper. "For the v1 version of IAI MovieBot, please cite" and CIKM'20 demo paper.

### PR DESCRIPTION
**Pull request by NoB0**
Fixes #237

Follow up issues were created to address TODOs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,42 +6,95 @@
...
 ![Python version](https://img.shields.io/badge/python-3.9-blue)
 
-IAI MovieBot is a conversational recommender system for movies.  It follows a standard task-oriented dialogue system architecture, comprising of natural language understanding (NLU), dialogue manager (DM), and natural language generation (NLG) components.  The distinctive features of IAI MovieBot include a task-specific dialogue flow, a multi-modal chat interface, and an effective way to deal with dynamically changing user preferences.  While our current focus is limited to movies, the system aims to be a reusable development framework that can support users in accomplishing recommendation-related goals via multi-turn conversations.
+IAI MovieBot is a conversational recommender system for movies. It follows a standard task-oriented dialogue system architecture, comprising of natural language understanding (NLU), dialogue manager (DM), and natural language generation (NLG) components. Additionally, it includes a user model and a recommendation engine. Some modules may use different models, which can be trained using the provided training utilities. It also comes with different deployment platforms.
```
Comment (kbalog) [inline README.md:9]: Nit (be more specific): "... different deployment platforms (such as XXX and YYY)"

File: docs/source/index.rst [hunk 2]
```diff
@@ -1,7 +1,8 @@
 IAI MovieBot |release| documentation
 =====================================
 
-IAI MovieBot is a conversational recommender system for movies.  It follows a standard task-oriented dialogue system architecture, comprising of natural language understanding (NLU), dialogue manager (DM), and natural language generation (NLG) components.  The distinctive features of IAI MovieBot include a task-specific dialogue flow, a multi-modal chat interface, and an effective way to deal with dynamically changing user preferences.  While our current focus is limited to movies, the system aims to be a reusable development framework that can support users in accomplishing recommendation-related goals via multi-turn conversations.
+IAI MovieBot is a conversational recommender system for movies. It follows a standard task-oriented dialogue system architecture, comprising of natural language understanding (NLU), dialogue manager (DM), and natural language generation (NLG) components. Additionally, it includes a user model and a recommendation engine. Some modules may use different models, which can be trained using the provided training utilities. It also comes with different deployment platforms.
```
Comment (kbalog) [inline docs/source/index.rst:4]: Same comment for platforms as above.

**Review by kbalog — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (NoB0) [reply README.md]: Done.

File: docs/source/index.rst (hunk 2 above)
Comment (NoB0) [reply docs/source/index.rst]: Done.

**Pull request merged**